src/firefly/js/ui/FileUploadProcessor.jsx

```diff
@@ -49,31 +49,39 @@ const tableWarning = 'Only loading the image(s), ignoring any selected table(s).
 
 export function resultSuccess(request) {
 
-    const {currentReport, currentDetailsModel, summaryModel, groupKey, acceptList, uniqueTypes}= request.additionalParams ?? {};
+    const {report, detailsModel, summaryModel, groupKey, acceptList, uniqueTypes,
+        acceptOneItem}= request.additionalParams ?? {};
     const summaryTblId = groupKey; //FileUploadAnalysis
     const fileCacheKey = getFileCacheKey(groupKey);
 
+    //highlighted row and entryType is used when acceptOneItem === true
+    const highlightedRowIdx = summaryModel?.highlightedRow;
+    const highlightedRow = summaryModel?.tableData?.data?.[highlightedRowIdx];
+    const entryType = highlightedRow?
+        highlightedRow[1]: summaryModel?.tableData?.data?.[0]?.[1];
+
     //determine if the file type or selections are valid to be loaded
-    const {valid, errorMsg, title} = determineValidity(acceptList, uniqueTypes, summaryModel, summaryTblId, currentReport);
+    const {valid, errorMsg, title} = determineValidity(acceptList, uniqueTypes, summaryModel,
+        summaryTblId, report, acceptOneItem, highlightedRowIdx, entryType);
 
     if (!valid) {
         showInfoPopup(errorMsg, title);
         return false;
     }
 
     const {loadType, tableIndices, imageIndices} = determineLoadType(acceptList, uniqueTypes, summaryModel,
-        summaryTblId, currentReport, fileCacheKey, request, currentDetailsModel);
+        summaryTblId, report, fileCacheKey, request, detailsModel, acceptOneItem, highlightedRowIdx, entryType);
 
     switch(loadType) {
         case LOAD_REGION:
-            sendRegionRequest(fileCacheKey, currentReport);
+            sendRegionRequest(fileCacheKey, report);
             return true;
 
         case LOAD_MOC:
             const mocMeta= {[MetaConst.PREFERRED_HIPS]: getAppHiPSForMoc()};
             if (request.mocOp==='table') mocMeta[MetaConst.IGNORE_MOC]='true';
             //loadToUI = true if request.mocOp==='table', else loadToUI=false
-            sendTableRequest(tableIndices, fileCacheKey, Boolean(request.tablesAsSpectrum==='spectrum'), currentReport, Boolean(request.mocOp==='table'), mocMeta);
+            sendTableRequest(tableIndices, fileCacheKey, Boolean(request.tablesAsSpectrum==='spectrum'), report, Boolean(request.mocOp==='table'), mocMeta);
```

Comment:
I did not catch this in the last PR, you don't need to wrap a `===` with Boolean, it is redundant. The result of a `===` is a boolean. You should fix that though out the swtich.